src/v/storage/segment_utils.cc

```diff
@@ -597,10 +597,18 @@ ss::future<ss::lw_shared_ptr<segment>> make_concatenated_segment(
               "Aborting compaction of closed segment: {}", *segment));
         }
     }
+
     auto compacted_idx_path = compacted_index_path(path);
+    if (co_await ss::file_exists(compacted_idx_path.string())) {
+        co_await ss::remove_file(compacted_idx_path.string());
```

Comment:
I'm sure this is correct, but the code reads very strangely like: (1) create file (2) immediately remove file.

EDIT: oh, never mind. it's calculating the path and removing a file with the same path if it happens to exist?

Reply:
>  it's calculating the path and removing a file with the same path if it happens to exist?

yeah, exactly